Rlog/ViewModel/WorkSpace/WorkSpaceCreateCreatingScheduleViewModel.swift

```diff
@@ -9,8 +9,157 @@ import SwiftUI
+                startMinute = oldValue
+                errorMessage = "59분을 초과한 값을 넣을 수 없습니다."
+                isShowingConfirmButton = false
+            } else {
+                errorMessage = ""
+            }
+        }
+    }
+    
+    @Published var endHour = "" {
+        didSet {
+            guard let endHour = Int(endHour) else { return }
+            if endHour > 24 {
+                self.endHour = oldValue
+                errorMessage = "24시간을 초과한 값을 넣을 수 없습니다."
+                isShowingConfirmButton = false
+            } else if Int(startHour) ?? 0 > endHour {
+                errorMessage = "출근시간 보다 퇴근시간이 빠릅니다"
+                isShowingConfirmButton = false
+            } else {
+                errorMessage = ""
+                checkAllInputFilled()
+            }
+        }
+    }
+    
+    @Published var endMinute = "" {
+        didSet {
+            if Int(endMinute) ?? 0 > 59 {
+                endMinute = oldValue
+                errorMessage = "59분을 초과한 값을 넣을 수 없습니다."
+                isShowingConfirmButton = false
+            } else {
+                errorMessage = ""
+            }
+        }
+    }
+    
+    init(isShowingModal: Binding<Bool>, scheduleList: Binding<[ScheduleModel]>) {
+        self._isShowingModal = isShowingModal
+        self._scheduleList = scheduleList
+    }
+    
+    func didTapDayPicker(index: Int) {
+        sevenDays[index].isSelected.toggle()
+        checkAllInputFilled()
+    }
+    
+    func didTapConfirmButton() {
+        //MARK: 어떤 코드인지 확인해보기
+        // -> 바인딩 해서 처리하는데 비동기처리가 필요할까?
+        Task {
+            await appendScheduleToList()
+            await dismissModal()
+        }
+    }
 }
 
+// MARK: - Private Functions
+private extension WorkSpaceCreateCreatingScheduleViewModel {
+    @MainActor
+    func appendScheduleToList() async {
+        if startMinute.isEmpty {
+            startMinute = "00"
+        }
+        if endMinute.isEmpty {
+            endMinute = "00"
+        }
+        scheduleList.append(
+            ScheduleModel(
+                repeatedSchedule: self.getDayList(),
+                startHour: self.startHour,
+                startMinute: self.startMinute,
+                endHour: self.endHour,
+                endMinute: self.endMinute
+            )
+        )
+    }
+    
+    @MainActor
+    func dismissModal() {
+        isShowingModal = false
+    }
+    
+    func getDayList() -> [String] {
+        var dayList: [String] = []
+        for day in sevenDays {
+            if day.isSelected {
+                dayList.append(day.dayName)
+            }
+        }
+        return dayList
+    }
+    
+    func checkAllInputFilled() {
+        if !startHour.isEmpty && !endHour.isEmpty && startHour != endHour && !getDayList().isEmpty {
+            isShowingConfirmButton = true
+            return
+        }
+        isShowingConfirmButton = false
+    }
+}
+
+struct selectedDayModel: Hashable {
```

Comment:
```suggestion
struct SelectedDayModel: Hashable {
```

이 Model은 ViewModel에서만 사용되는걸까요 ? !